manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml

```diff
@@ -1,7 +1,11 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
+{{- if not .Values.global.externalIstiod }}
   name: istiod-cr{{- if not (eq .Values.revision "")}}-{{ .Values.revision }}{{- end }}{{ .Release.Namespace }}
+{{- else }}
+  name: istiod{{- if not (eq .Values.revision "")}}-{{ .Values.revision }}{{- end }}-{{ .Values.global.istioNamespace }}
```

Comment:
Why don't we want `-cr` here?  Same question in a few places. Seems odd to make this inconsistent.

Reply:
The `-cr` is there to avoid name conflict with base charts which causes Helm install to fail (we just moved a bunch of resources out of base but have to keep them around for a while so they don't get pruned on upgrade).

Reply:
@Monkeyanator In that case, it sounds like it would be OK to use the `-cr` name for both the local and external istiod cases (and maybe get rid of all the if/else blocks in the template).

The other difference is that the local case appends `{{ .Release.Namespace }}` while the remote one uses `-{{ .Values.global.istioNamespace }}`. Two questions:

1. Why is there no `-` character in the local case. Is that a bug? Shouldn't it be `-{{ .Release.Namespace }}`?
2. Could we change it to `.Values.global.istioNamespace` for both, and get rid of the if/else blocks?

Reply:
1. Yep that looks like a typo 
2. I think that should be fine, not sure if there's a case where the two would be different

Reply:
Thanks, I'll try a bit of cleanup in a following PR.